$OpenBSD: patch-imap_tls_c,v 1.5 2020/05/14 12:26:39 ajacoutot Exp $

XXX check at next update
LIBRESSL_HAS_TLS1_3 is not yet defined on LibreSSL

Index: imap/tls.c
--- imap/tls.c.orig
+++ imap/tls.c
@@ -798,7 +798,7 @@ EXPORTED int     tls_init_serverengine(const char *ide
     const char *tls_versions = config_getstring(IMAPOPT_TLS_VERSIONS);
 
     if (strstr(tls_versions, "tls1_3") == NULL) {
-#if (OPENSSL_VERSION_NUMBER >= 0x1010100fL)
+#if (OPENSSL_VERSION_NUMBER >= 0x1010100fL) && !defined(LIBRESSL_VERSION_NUMBER)
         //syslog(LOG_DEBUG, "TLS server engine: Disabled TLSv1.3");
         off |= SSL_OP_NO_TLSv1_3;
 #else
